sunpy/image/transform.py

```diff
@@ -130,20 +128,7 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
                           RuntimeWarning)
             adjusted_image = np.nan_to_num(adjusted_image)
```

Comment:
I think in general fixing input for users is to be avoided.  This is because we can't really anticipate what the user intends.  Pixels that are NaNs or infs can be replaced in a multitude of different ways.